src/engine/controls/cuecontrol.cpp

```diff
@@ -594,14 +617,19 @@ void CueControl::hotcueSet(HotcueControl* pControl, double v) {
     pCue->setHotCue(hotcue);
     pCue->setLabel("");
     pCue->setType(Cue::Type::HotCue);
-    // TODO(XXX) deal with spurious signals
-    attachCue(pCue, pControl);
 
-    if (getConfig()->getValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false)) {
-        const QList<PredefinedColorPointer> predefinedColors = Color::kPredefinedColorsSet.allColors;
-        pCue->setColor(predefinedColors.at((hotcue % (predefinedColors.count() - 1)) + 1));
+    auto hotcueColorPalette = m_colorPaletteSettings.getHotcueColorPalette();
+    ConfigKey autoHotcueColorsKey("[Controls]", "auto_hotcue_colors");
+    if (getConfig()->getValue(autoHotcueColorsKey, false)) {
+        auto colors = hotcueColorPalette.m_colorList;
+        pCue->setColor(colors.at(hotcue % colors.count()));
+    } else {
+        pCue->setColor(hotcueColorPalette.m_colorList.first());
```

Comment:
Determining and setting the color should be done separately. What this code does is actually very simple, but it is not concise the way it is written.

Reply:
This code will be removed anyway.